tests/foreman/ui_airgun/test_subscription.py

```diff
@@ -0,0 +1,309 @@
+
+    :id: 580fc072-01e0-4f83-8cbb-2a8522d76243
+
+    :expectedresults:
+        1. The manifest was uploaded successfully.
+        2. When attempting to delete the manifest the confirmation dialog contains informative
+            message which warns user about downsides and consequences of manifest deletion.
+        3. When hitting cancel the manifest was not deleted.
+        4. When deleting and confirming deletion, the manifest was deleted successfully.
+
+    :BZ: 1266827
+
+    :CaseImportance: Critical
+    """
+    expected_message_lines = [
+        'Are you sure you want to delete the manifest?',
+        'Note: Deleting a subscription manifest is STRONGLY discouraged. '
+        'Deleting a manifest will:',
+        'Delete all subscriptions that are attached to running hosts.',
+        'Delete all subscriptions attached to activation keys.',
+        'Disable Red Hat Insights.',
+        'Require you to upload the subscription-manifest and re-attach '
+        'subscriptions to hosts and activation keys.',
+        'This action should only be taken in extreme circumstances or for '
+        'debugging purposes.',
+    ]
+    org = entities.Organization().create()
+    _, temporary_local_manifest_path = mkstemp(prefix='manifest-', suffix='.zip')
+    with manifests.clone() as manifest:
+        with open(temporary_local_manifest_path, 'wb') as file_handler:
+            file_handler.write(manifest.content.read())
+
+    with session:
+        session.organization.select(org.name)
+        session.subscription.add_manifest(temporary_local_manifest_path)
+        assert session.subscription.has_manifest
+        delete_message = session.subscription.read_delete_manifest_message()
+        assert ' '.join(expected_message_lines) == delete_message
+        assert session.subscription.has_manifest
+        session.subscription.delete_manifest()
+        assert not session.subscription.has_manifest
+
+
+@tier2
+def test_positive_access_with_non_admin_user_without_manifest(test_name):
+    """Access subscription page with user that has only view_subscriptions
+    permission and organization that has no manifest uploaded.
+
+    :id: dab9dc15-39a8-4105-b7ff-ecef909dc6e6
+
+    :expectedresults: Subscription page is rendered properly without errors
+
+    :BZ: 1417082
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    org = entities.Organization().create()
+    role = entities.Role(organization=[org]).create()
+    create_role_permissions(
+        role,
+        {'Katello::Subscription': ['view_subscriptions']}
+    )
+    user_password = gen_string('alphanumeric')
+    user = entities.User(
+        admin=False,
+        role=[role],
+        password=user_password,
+        organization=[org],
+        default_organization=org,
+    ).create()
+    with Session(test_name, user=user.login, password=user_password) as session:
+        session.subscription.navigate_to(session.subscription, 'All')
```

Comment:
I left access to `navigate_to` from tests layer just in case for corner cases when we need only navigation and can't use smth else. But yeah, consider using it a bad tone if some other options are available.

Reply:
yeah, please go ahead with search approach. It will probably return empty table, but that is fine